src/Components/Card.js

```diff
@@ -69,14 +67,20 @@ const Card = ({
           ) : (
             ''
           )}
-          <Button>
-            <Link
-              onClick={() => setCardIsInDetailsMode(true)}
-              to={`/details/${id}`}
-            >
+          {bookmarking ? (
+            <Button onClick={() => setCardIsInDetailsMode(true)}>
               Show more
-            </Link>
-          </Button>
+            </Button>
+          ) : (
+            <Button>
+              <Link
```

Comment:
Instead of using a Link, your Card component could use an optional onClickCallback property, which only your MainPage provides. In the MainPage you could define it similar to this:
```
import { useNavigate } from 'react-router-dom';
...
const navigate = useNavigate();
...
<Card
   ...
   onClickCallback={() => navigate(`details/${character.id}`)}
/>
```

Reply:
..also the id prop of your Card component is unused

Reply:
...also the Bookmark component could use a property "bookmarked" instead of setting a className. Styled components are able to use this prop, like this:
  `background-color: ${props => props.bookmarked ? 'black' : 'rgba(15,15,13,0.3)'};`

Reply:
Thanks again for this feedback, Thomas. I learned a lot from it! 
Changed all the above.